pkg/cmd/build/depot.go

```diff
@@ -0,0 +1,173 @@
+		okCtx:   okCtx,
+		fs:      afero.NewOsFs(),
+
+		newDepotBuild: build.NewBuild,
+		acquireMachine: func(ctx context.Context, buildId, token, platform string) (depotMachineConnector, error) {
+			return machine.Acquire(ctx, buildId, token, platform)
+		},
+	}
+}
+
+// release releases the depot's machine and finishes the build.
+func (db *depotBuilder) release(build build.Build) {
+	build.Finish(db.err)
+
+	err := db.machine.Release()
+	if err != nil {
+		db.ioCtrl.Logger().Infof("failed to release depot's machine: %s", err)
+	}
+}
+
+type runAndHandleBuildFn func(ctx context.Context, c *client.Client, opt *client.SolveOpt, buildOptions *types.BuildOptions, okCtx OktetoContextInterface, ioCtrl *io.IOController) error
+
+func (db *depotBuilder) Run(ctx context.Context, buildOptions *types.BuildOptions, run runAndHandleBuildFn) error {
+	db.ioCtrl.Logger().Info("building your image on depot's machine")
+
+	// Register a new build.
+	req := &cliv1.CreateBuildRequest{
+		ProjectId: db.project,
+		Options: []*cliv1.BuildOptions{
+			{
+				Command: cliv1.Command_COMMAND_BUILD,
+				Tags:    []string{buildOptions.Tag},
+				Push:    true,
+				Load:    true,
+			},
+		},
+	}
+
+	build, err := db.newDepotBuild(ctx, req, db.token)
+	if err != nil {
+		return err
+	}
+	if buildOptions.Platform == "" {
+		buildOptions.Platform = defaultPlatform
+	}
+
+	db.machine, err = db.acquireMachine(ctx, build.ID, build.Token, buildOptions.Platform)
+	if err != nil {
+		return err
+	}
+
+	client, err := db.getBuildkitClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	if buildOptions.File != "" {
+		buildOptions.File, err = GetDockerfile(buildOptions.File, db.okCtx)
+		if err != nil {
+			return err
+		}
+		defer func() {
+			err := db.fs.Remove(buildOptions.File)
+			if err != nil {
+				db.ioCtrl.Logger().Infof("failed to remove file after build process: %s", err)
+			}
+		}()
+	}
+
+	// create a temp folder - this will be remove once the build has finished
+	secretTempFolder, err := createSecretTempFolder()
+	if err != nil {
+		return err
+	}
+	defer func() {
+		err := db.fs.RemoveAll(secretTempFolder)
+		if err != nil {
+			db.ioCtrl.Logger().Infof("failed to remove temp secrets folder after build process: %s", err)
+		}
+	}()
+
+	opt, err := getSolveOpt(buildOptions, db.okCtx, secretTempFolder, db.fs)
+	if err != nil {
+		return fmt.Errorf("failed to create build solver: %w", err)
+	}
+
+	db.ioCtrl.Logger().Info("Acquiring a depot's machine..")
+
+	defer db.release(build)
+	db.ioCtrl.Logger().Infof("building '%s' on depot's machine..", buildOptions.Tag)
+
+	return run(ctx, client, opt, buildOptions, db.okCtx, db.ioCtrl)
+}
+
+// getBuildkitClient returns a buildkit client connected to the depot's machine.
+func (db *depotBuilder) getBuildkitClient(ctx context.Context) (*buildkitClient.Client, error) {
+	// Check buildkitd readiness. When the buildkitd starts, it may take
+	// quite a while to be ready to accept connections when it loads a large boltdb.
```

Comment:
what is "boltdb"?

Reply:
according to the depot go sdk documentation, the buildkit daemon can take a while to accept connections as it loads a key value database, this is a boltdb database

Reply:
thanks for clarifying, I've never heard of it before :D